content/src/main/content/jcr_root/apps/acs-commons/touchui-widgets/showhidedialogfields/source/showhidedialogfieldstabs.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Extension to the standard dropdown/select and checkbox component. It enables hidding/unhidding of multiple dialog fields 
+ * and dialog tabs based on the selection made in the dropdown/select or on checkbox check or their combination. 
+ *
+ * How to use:
+ * - add the empty property cq-dialog-dropdown-checkbox-showhide to the dropdown/select or checkbox element
+ * - add the data attribute cq-dialog-dropdown-checkbox-showhide-target to the dropdown/select or checkbox element, 
+ *   value should be the selector, usually a specific class name, to find all possible target elements that can be shown/hidden.
+ * - add the target class to each target component that can be shown/hidden
+ * - add the class hide to each target component to make them initially hidden
+ * - add the attribute dropdownshowhidetargetvalue to the target component, the value should equal the value of the select
+ *   option that will unhide this element. Multiple values can be provided separated with spaces.
+ * - add the attribute checkboxshowhidetargetvalue to the target component, the value should equal to:

```

Comment:
there's a lack of symmetry with the property/attribute names here. I'd prefer that they were a consistent style.

Reply:
Also, is it really necessary to differentiate if the value came from a checkbox vs. a dropdown? Seems like it would be simpler to just have a single attribute which held the value (or a comma-delimited list of values) and if one of the matches either the dropdown value or the checkbox value.

But, TBH, I don't know the use case well enough to judge. Just looking at this from an API standpoint.

Reply:
@justinedelson It is necessary to differentiate between the checkbox and dropdown because of use case where we would want to display a field when a certain dropdown value is selected and the checkbox is supposed to be checked. If we use the same attribute name for the checkebox value e.g. [dropdown-val1, dropdown-val2, checked/true], the first expectation from it would be that if either of those values is selected, then the field would be shown and not the case the where combination of dropdown value and check box is checked.